### ISSUE TITLE
[java] int + char Possible Intention for String Appending

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#2852]
**Proposed Rule Name:** UnintentionalAddition

**Proposed Category:** Error Prone

**Description:**

Please create a rule that checks if two variables are added together and then converted into a `String` via `+`.

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

The following code adds the `char` and `int` together and then converts the result into a `String`.  The result being `66 employee`.  This is probably not what the programmer intended.
```
char type;
int level;

type = 'A';
level = 1;
text = type + level + " employee";
```

The following code fixes the problem.  The result being `A1 employee`.
```
char type;
int level;

type = 'A';
level = 1;
text = Character.toString(type) + level + " employee";
```

The following makes it clearer that the programmer intends to add the `char` and `int` and then convert into a `String`.
```
char type;
int level;

type = 'A';
level = 1;
text = (type + level) + " employee";
```

The following makes it very clear that the programmer intends to add the `char` and `int` and then convert into a `String`.
```
char type;
int level, typeLevel;

type = 'A';
level = 1;
typeLevel = type + level;
text = typeLevel + " employee";
```

**Possible Properties:**

I cannot think of any properties.  Can you?
